config/routes.rb

```diff
@@ -1,11 +1,12 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-  root 'users#index'
-  # get 'about', to: 'pages#about'
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  get '/', to: 'users#index'
 
-  resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+  resources :users, only: %i[index show] do
+    resources :posts, only: %i[index show new create] do
+      resources :comments, only: [:create]
+    end
   end
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+
+  put 'user/:user_id/post/:id/like', to: 'posts#like', as: 'like'
```

Comment:
The method for creating a new like should be `POST` not `PUT`. Kindly use `POST` for the likes route.